src/ssh/sftp_client.cpp

```diff
@@ -154,25 +142,43 @@ void mp::SFTPClient::pull_file(const std::string& source_path, const std::string
     } while (total < size);
 }
 
-void mp::SFTPClient::stream_file(const std::string& destination_path)
+void mp::SFTPClient::stream_file(const std::string& destination_path, std::istream& cin)
 {
     auto full_destination_path = full_destination(destination_path, stream_file_name);
-    SFTPSessionUPtr sftp{make_sftp_session(*ssh_session)};
-    SSH::throw_on_error(sftp, *ssh_session, "[sftp stream] init session failed", sftp_init);
-
     SFTPFileUPtr file_handle{
         sftp_open(sftp.get(), full_destination_path.c_str(), O_WRONLY | O_CREAT | O_TRUNC, file_mode), sftp_close};
     SSH::throw_on_error(sftp, *ssh_session, "[sftp stream] open failed", sftp_get_error);
 
-    QTextStream in_stream(stdin);
-    QString text;
-    QByteArray data;
+    std::array<char, max_transfer> data;
+    while (!cin.eof())
+    {
+        cin.read(data.data(), data.size());
```

Comment:
Thank you!

Reply:
Noticeable faster too. I guess QTesxStream was slow as it was processing the stream..